test/distributed/_tensor/test_dtensor.py

```diff
@@ -55,6 +62,52 @@ def test_dtensor_constructor(self):
                 requires_grad=True,
             )
 
+    @with_comms
+    def test_meta_dtensor(self):
+        device_mesh = self.build_device_mesh()
+        dist_specs = [[Shard(0)], [Replicate()]]
+        meta_tensor = torch.randn(1024, 2048, device="meta")
+        for dist_spec in dist_specs:
+            # Test distribute_tensor on meta tensor
+            meta_dtensor = distribute_tensor(meta_tensor, device_mesh, dist_spec)
+            self.assertTrue(meta_dtensor.is_meta)
+            meta_dtensor = torch.empty_like(meta_dtensor, device=self.device_type)
+            torch.nn.init.constant_(meta_dtensor, 1.2)
+            self.assertFalse(meta_dtensor.is_meta)
```

Comment:
If this tensor is not a meta dtensor, shall we use a different variable name?

Reply:
What do you mean? This is essentially a tensor attribute: https://pytorch.org/docs/stable/generated/torch.Tensor.is_meta.html.